jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -41,17 +44,22 @@
    * @return a new {@link DockerClient}
    */
   public static DockerClient newClient() {
-    return new DockerClient(defaultProcessBuilderFactory(DEFAULT_DOCKER_CLIENT));
+    return newClient(null, null);
```

Comment:
That works, but the downside is that `BuildDockerTask` will have to do

```java
if (dockerExecutable != null && dockerEnvironment != null) {
  ... = DockerClient.newClient(dockerExecutable, dockerEnvironment);
} else if (dockerExecutable == null && dockerEnvironment == null ) {
  ... = DockerClient.newClient();
} else {
  ... ???
}
```

or make internal defaults of `DockerClient` public, so that

```java
... = DockerClient.newClient(
    dockerClient == null ? DockerClient.DEFAULT_DOCKER_CLIENT : dockerClient.toString(),
    dockerEnvironment == null ? ... : dockerEnvironment);
```

or define `DockerBuildTask`s own defaults, duplicating defaults in multiple places.

Reply:
Hmm, maybe this warrants a simple `Builder` to make the construct clearer - using `null` to mean default where the default is not `null` feels a bit off

Reply:
Good idea. Let me try a builder.